### ISSUE TITLE
[BUG] Compute metadata orientation edge case 

### ISSUE THREAD

**Issue by benjaminpkane** [voxel51/fiftyone#2047]
A user has encountered an issue when using `compute_metadata()` in which the exif orientation is not accounted for and the width and height are flipped.

Perhaps `compute_metadata()` should account for orientation as browser rendering will, I believe.

**Comment by brimoor** [comment]
Relevant details:

- `compute_metadata()` uses `PIL.Image.open()` to compute height/width:
https://github.com/voxel51/fiftyone/blob/7304b69abb556e8899084329d9dde3197e1913b8/fiftyone/core/metadata.py#L273-L284
- The rest of the library uses some mix of `PIL.Image` and `cv2.read` to read images whenever a builtin method requests this:
https://github.com/voxel51/fiftyone/blob/7304b69abb556e8899084329d9dde3197e1913b8/fiftyone/utils/torch.py#L1592
https://github.com/voxel51/eta/blob/9a89132a0fc8e8d519faf1fbb71f93981f97255d/eta/core/image.py#L460

The issue below suggests that OpenCV may not have supported EXIF until pretty recently (and also that PNG itself didn't support EXIF until 2017):
https://github.com/opencv/opencv/issues/16579

I suspect it may be difficult to pin down our browser + `PIL` + `cv2` dependencies in order to uniformly respect or ignore EXIF in a reliable way...

**TLDR**: working with EXIF oriented images in FiftyOne is definitely "use at your own risk" territory.

**Comment by mthatfalvi** [comment]
I would propose 2 ways to fix this as this has been an issue for me. 
1. To the two functions above add the following line
`from PIL import ImageOps
 img = Image.open(image_path)
img = ImageOps.exif_transpose(img)` 
This line will transpose the image if it needs to be according to the exif data else just returns a copy of the image. https://pillow.readthedocs.io/en/stable/reference/ImageOps.html
2. for cv2 functions the flag IMREAD_UNCHANGED which can be used as default if force_rgb is not true will not orient the image according to the exif data as describe [here](https://robototem.medium.com/opencv-imread-method-imread-unchanged-flag-is-explained-f2d5e52720b7). Either forcing the rotation buy setting force_rgb == true or another solution has to be found for opencv when using the alpha channel and exif orientation.
